build/docker/services/base/container-install-htcondor.sh

```diff
@@ -0,0 +1,138 @@
+    fi
+    $yum install epel-release
+    if [[ $OS_VERSION_ID == 7 ]]; then
+        $yum install yum-plugin-priorities
+    elif [[ $OS_VERSION_ID == 8 ]]; then
+        $yum install dnf-plugins-core
+        # enable CentOS PowerTools repo (whose name changed between CentOS releases)
+        (dnf config-manager --set-enabled powertools || dnf config-manager --set-enabled PowerTools)
+    fi
+    $yum install "${repo_base_centos}/${HTCONDOR_SERIES}/htcondor-release-current.el${OS_VERSION_ID}.noarch.rpm"
+    rpm --import /etc/pki/rpm-gpg/*
+    if [[ $HTCONDOR_VERSION == daily ]]; then
+        if [[ $OS_VERSION_ID == 7 ]]; then
+            yum-config-manager --enable htcondor-daily
+        elif [[ $OS_VERSION_ID == 8 ]]; then
+            dnf config-manager --set-enabled htcondor-daily
+        fi
+    fi
+
+    $yum install "${extra_packages_centos[@]}"
+    rpm -q "${extra_packages_centos[@]}"
+    if [[ $HTCONDOR_VERSION == latest || $HTCONDOR_VERSION == daily ]]; then
+        $yum install condor
+    else
+        $yum install "condor = ${HTCONDOR_VERSION}"
+    fi
+    rpm -q condor
+
+    yum clean all
+    rm -rf /var/cache/yum/*
+
+# Ubuntu ---------------------------------------------------------------------
+
+elif [[ $OS_ID == ubuntu ]]; then
+    apt_install="apt-get install -qy"
+    export DEBIAN_FRONTEND=noninteractive
+
+    apt-get update -q
+    if $update_os; then
+        apt-get upgrade -qy
+    fi
+    $apt_install gnupg2 wget
+
+    # TODO Is there a per-series key I should be using instead?
+    set -o pipefail
+    wget -qO - "${repo_base_ubuntu}/HTCondor-Release.gpg.key" | apt-key add -
+    set +o pipefail
+    echo "deb     ${repo_base_ubuntu}/${HTCONDOR_SERIES}/${OS_UBUNTU_CODENAME} ${OS_UBUNTU_CODENAME} contrib" >> /etc/apt/sources.list
+    echo "deb-src ${repo_base_ubuntu}/${HTCONDOR_SERIES}/${OS_UBUNTU_CODENAME} ${OS_UBUNTU_CODENAME} contrib" >> /etc/apt/sources.list
+
+    apt-get update -q
+    $apt_install "${extra_packages_ubuntu[@]}"
+    if [[ $HTCONDOR_VERSION == latest || $HTCONDOR_VERSION == daily ]]; then
+        $apt_install htcondor
+    else
+        # Get the version-release string because unlike RHEL I can't install with just the version
+        htcondor_version_release=$(apt-cache madison htcondor | grep -F "${HTCONDOR_VERSION}" | awk '/Packages/ {print $3; exit}')
+        $apt_install "htcondor=$htcondor_version_release"
+    fi
+
+    rm -rf /var/lib/apt/lists/* /var/cache/apt/*
+
+fi
+
+# Common ---------------------------------------------------------------------
+
+# Create passwords directories for token or pool password auth.
+#
+# Only root needs to know the pool password but other condor daemons
+# need access to the tokens.
+#
+# TODO: Do the packages take care of this now?
+install -m 0700 -o root -g root -d /etc/condor/passwords.d
+install -m 0700 -o condor -g condor -d /etc/condor/tokens.d
```

Comment:
The tokens.d directory should be owned by root.
```suggestion
install -m 0700 -o root -g root -d /etc/condor/tokens.d
```

Reply:
Unless that changed recently, the condor user needs to be able to read the tokens. I could have this be owned by `root:condor` and have `0750` perms if you'd prefer that...

Reply:
This is changing in 8.9.12. See HTCONDOR-266.

Reply:
OK. Since 8.9.12 isn't out yet and we might want to use this to (re)build images for earlier condor versions, I will put this behind a version check.

Reply:
Please resolve this soon. I want to be able to build images on release day next week.

Reply:
Fixed by a77531e